Core/GameEngine/Include/GameNetwork/NetPacketStructs.h

```diff
@@ -0,0 +1,405 @@
+struct NetPacketRouterAckCommand {
+	NetPacketCommandTypeField commandType;
+	NetPacketRelayField relay;
+	NetPacketPlayerIdField playerId;
+	NetPacketDataFieldHeader dataHeader;
+};
+
+////////////////////////////////////////////////////////////////////////////////
+// Disconnect Vote Command Packet
+////////////////////////////////////////////////////////////////////////////////
+
+// Disconnect vote command
+// Fields: T + type, R + relay, P + playerID, C + commandID, D + slot + voteFrame
+struct NetPacketDisconnectVoteCommand {
+	NetPacketCommandTypeField commandType;
+	NetPacketRelayField relay;
+	NetPacketPlayerIdField playerId;
+	NetPacketCommandIdField commandId;
+	NetPacketDataFieldHeader dataHeader;
+	UnsignedByte slot;
+	UnsignedInt voteFrame;
+};
+
+////////////////////////////////////////////////////////////////////////////////
+// Packed Structs for getPackedByteCount() calculations
+// These structs represent the fixed portion of variable-length command messages
+////////////////////////////////////////////////////////////////////////////////
+
+struct NetPacketChatCommand {
+	NetPacketCommandTypeField commandType;
+	NetPacketRelayField relay;
+	NetPacketPlayerIdField playerId;
+	NetPacketDataFieldHeader dataHeader;
+	NetPacketFrameField frame;
+	NetPacketCommandIdField commandId;
+};
+
+struct NetPacketDisconnectChatCommand {
+	NetPacketCommandTypeField commandType;
+	NetPacketRelayField relay;
+	NetPacketPlayerIdField playerId;
+	NetPacketDataFieldHeader dataHeader;
+};
+
+// Game command packed struct (variable: game message data follows)
+struct NetPacketGameCommand {
+	NetPacketCommandTypeField commandType;
+	NetPacketFrameField frame;
+	NetPacketRelayField relay;
+	NetPacketPlayerIdField playerId;
+	NetPacketCommandIdField commandId;
+	NetPacketDataFieldHeader dataHeader;
+};
+
+// Wrapper command packet (fixed size - contains metadata about wrapped command)
+// Fields: T + type, P + playerID, C + commandID, R + relay, D + metadata
+struct NetPacketWrapperCommand {
+	NetPacketCommandTypeField commandType;
+	NetPacketPlayerIdField playerId;
+	NetPacketCommandIdField commandId;
+	NetPacketRelayField relay;
+	NetPacketDataFieldHeader dataHeader;
+	UnsignedShort wrappedCommandId;
+	UnsignedInt chunkNumber;
+	UnsignedInt numChunks;
+	UnsignedInt totalDataLength;
+	UnsignedInt dataLength;
+	UnsignedInt dataOffset;
+};
+
+// File command packed struct (variable: filename and file data follow)
+struct NetPacketFileCommand {
+	NetPacketCommandTypeField commandType;
+	NetPacketRelayField relay;
+	NetPacketPlayerIdField playerId;
+	NetPacketDataFieldHeader dataHeader;
+	NetPacketCommandIdField commandId;
+};
+
+// File announce command packed struct (variable: filename and metadata follow)
+struct NetPacketFileAnnounceCommand {
+	NetPacketCommandTypeField commandType;
+	NetPacketRelayField relay;
+	NetPacketPlayerIdField playerId;
+	NetPacketDataFieldHeader dataHeader;
+	NetPacketCommandIdField commandId;
```

Comment:
Is this order correct?

Reply:
I don't think the order here changes what comes out over the wire, but I found 5 structs that had field orders out of sync with their respective wire formats. They all match now. NetPacketProgressMessage still had commandId which isn't used so I removed that - this also cleans up that weird subtraction we talked about above